lib/plugins/aws/invokeLocal/index.js

```diff
@@ -267,62 +259,121 @@ class AwsInvokeLocal {
       (this.options.functionObj.layers || this.serverless.service.provider.layers || []).map(
         layer => {
           if (layer.Ref) {
-            return layers[layer.Ref].path;
+            const targetLayer = layers[layer.Ref];
+
+            if (targetLayer.path) {
+              return targetLayer.path;
+            }
+            if (targetLayer.package && targetLayer.package.artifact) {
+              const layerArtifactContentPath = path.join('.serverless', 'layers', layer.Ref);
+              return dirExists(layerArtifactContentPath).then(exists => {
+                if (exists) {
+                  return layerArtifactContentPath;
+                }
+                this.serverless.cli.log(`Unziping ${layer.Ref}...`);
+                return decompress(targetLayer.package.artifact, layerArtifactContentPath).then(
+                  () => layerArtifactContentPath
+                );
+              });
+            }
           }
           const arnParts = layer.split(':');
           const layerArn = arnParts.slice(0, -1).join(':');
           const layerVersion = Number(arnParts.slice(-1)[0]);
           const layerContentsPath = path.join('.serverless', 'layers', arnParts[6], arnParts[7]);
           const layerContentsCachePath = path.join(cachePath, 'layers', arnParts[6], arnParts[7]);
-          if (fs.existsSync(layerContentsPath)) {
-            return layerContentsPath;
-          }
-          let downloadPromise = BbPromise.resolve();
-          if (!fs.existsSync(layerContentsCachePath)) {
-            this.serverless.cli.log(`Downloading layer ${layer}...`);
-            mkdirp.sync(path.join(layerContentsCachePath));
-            downloadPromise = this.provider
-              .request('Lambda', 'getLayerVersion', {
-                LayerName: layerArn,
-                VersionNumber: layerVersion,
-              })
-              .then(layerInfo =>
-                download(layerInfo.Content.Location, layerContentsCachePath, { extract: true })
-              );
-          }
-          return downloadPromise
-            .then(() => fse.copySync(layerContentsCachePath, layerContentsPath))
+
+          return dirExists(layerContentsPath)
+            .then(exists => {
+              if (exists) {
+                return layerContentsPath;
+              }
+              return dirExists(layerContentsCachePath);
+            })
+            .then(exists => {
```

Comment:
If I see correctly, this `then` should be attached to return of `dirExists` two lines above.